src/lake/Lake/Config/Env.lean

```diff
@@ -34,20 +37,26 @@ structure Env where
   initSharedLibPath : SearchPath
   /-- The initial binary search path of the environment (i.e., `PATH`). -/
   initPath : SearchPath
-  deriving Inhabited, Repr
+  deriving Inhabited
 
 namespace Env
 
 /-- Compute an `Lake.Env` object from the given installs and set environment variables. -/
 def compute (lake : LakeInstall) (lean : LeanInstall) (elan? : Option ElanInstall) : BaseIO Env :=
   return {
     lake, lean, elan?,
+    pkgUrlMap := ← computePkgUrlMap
     initToolchain := (← IO.getEnv "ELAN_TOOLCHAIN").getD ""
     initLeanPath := ← getSearchPath "LEAN_PATH",
     initLeanSrcPath := ← getSearchPath "LEAN_SRC_PATH",
     initSharedLibPath := ← getSearchPath sharedLibPathEnvVar,
     initPath := ← getSearchPath "PATH"
   }
+where
+  computePkgUrlMap := do
+    let some urlMapStr ← IO.getEnv "LAKE_PKG_URL_MAP" | return {}
+    let .ok urlMap := Json.parse urlMapStr |>.bind fromJson? | return {}
```

Comment:
This should error, rather than silently ignoring the bad JSON?

Reply:
This does mean that an improper environment will kill virtually any `lake` command which I was not sure was desirable. But it sounds like that is fine, so will do.

Reply:
I think given that JSON-via-environment-variable is pretty dodgy to begin with, failing fast is better than the user not noticing a stray comma and not understanding why they can't get their packages.